libphal/phal_dump.C

```diff
@@ -0,0 +1,299 @@
+	} catch (std::ofstream::failure& oe) {
+		int err = errno;
+		log(level::ERROR,
+		    "Failed to write to dump file err=%d error message=%s", err,
+		    oe.what());
+		throw dumpError_t(exception::FILE_OPERATION_FAILURE);
+	}
+	outfile.close();
+}
+
+void collectSBEDump(uint32_t id, uint32_t failingUnit,
+		    std::filesystem::path& dumpPath)
+{
+	struct pdbg_target* proc = NULL;
+
+	log(level::INFO,
+	    "Collecting SBE dump: path=%s, id=%d, chip position=%d",
+	    dumpPath.string().c_str(), id, failingUnit);
+
+	std::stringstream ss;
+	ss << std::setw(8) << std::setfill('0') << id;
+	std::string idStr = ss.str();
+
+	std::string baseFilename =
+	    idStr + ".0_" + std::to_string(failingUnit) + "_SbeData_p10_";
+	proc = preCollection(failingUnit);
+	fapi2::ReturnCode fapiRc;
+
+	// Collect SBE local register dump
+	std::vector<SBESCOMRegValue_t> sbeScomRegValue;
+	fapiRc = p10_sbe_localreg_dump(proc, true, sbeScomRegValue);
+	if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+		log(level::ERROR,
+		    "Failed in p10_sbe_localreg_dump for proc=%s, rc=0x%08X",
+		    pdbg_target_path(proc), fapiRc);
+	} else {
+
+		std::string dumpFilename =
+		    baseFilename + "p10_sbe_localreg_dump";
+		std::filesystem::path basePath = dumpPath / dumpFilename;
+
+		try {
+			writeDumpFile(
+			    reinterpret_cast<char*>(&sbeScomRegValue[0]),
+			    sizeof(SBESCOMRegValue_t) * sbeScomRegValue.size(),
+			    basePath);
+		} catch (dumpError_t& e) {
+			log(level::ERROR,
+			    "Error in writing p10_sbe_localreg_dump file "
+			    "errorMsg=%s",
+			    e.what());
+		}
+	}
+
+	// Dump contents of various PIB Masters and Slaves internal registers
+	std::vector<sRegV> pibmsRegSet;
+	std::vector<sReg> regs2Dump;
+	if (pibms_regs_2dump.size()) {
+		regs2Dump.insert(regs2Dump.end(), pibms_regs_2dump.begin(),
+				 pibms_regs_2dump.end());
+	}
+	if (regs2Dump.size()) {
+		sRegV regv;
+		for (auto& reg : regs2Dump) {
+			regv.reg.attr = reg.attr;
+			regv.reg.addr = reg.addr;
+			regv.reg.name = reg.name;
+			pibmsRegSet.push_back(regv);
+		}
+	}
+
+	fapiRc = p10_pibms_reg_dump(proc, pibmsRegSet);
+	if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+		log(level::ERROR,
+		    "Failed in p10_pibms_reg_dump for proc=%s, rc=0x%08X",
+		    pdbg_target_path(proc), fapiRc);
+	} else {
+		std::string dumpFilename = baseFilename + "p10_pibms_reg_dump";
+		std::filesystem::path basePath = dumpPath / dumpFilename;
+		try {
+			writeDumpFile(reinterpret_cast<char*>(&pibmsRegSet[0]),
+				      sizeof(sRegV) * pibmsRegSet.size(),
+				      basePath);
+		} catch (dumpError_t& e) {
+			log(level::ERROR,
+			    "Error in writing p10_pibms_reg_dump file, "
+			    "errorMsg=%s",
+			    e.what());
+		}
+	}
+
+	// Dump the PIBMEM Array based on starting and number of address
+	std::vector<array_data_t> pibmemContents;
+	bool eccEnable = false;
+	uint32_t pibmemDumpStartByte = 0;
+	// Number of bytes to be read from PIBMEM
+	uint32_t pibmemDumpNumOfByte = 0x7D400;
```

Comment:
static constexpr uint32_t pibmemDumpNumOfByte = 0x7D400; 

so that this variable is declared and defined only once